lib/scholar/stats.ex

```diff
@@ -0,0 +1,174 @@
+defmodule Scholar.Stats do
+  @moduledoc """
+  Statistical functions
+
+  All of the functions in this module are implemented as
+  numerical functions and can be JIT or AOT compiled with
+  any supported `Nx` compiler.
+  """
+
+  import Nx.Defn
+
+  general = [
+    axes: [
+      type: {:custom, Scholar.Options, :axes, []},
+      default: [0],
+      doc: """
+      Axis to calculate the operation. If set to `nil` then
+      the operation is performed on the whole tensor.
+      """
+    ],
+    keep_axes: [
+      type: :boolean,
+      default: false,
+      doc: "If set to true, the axes which are reduced are left."
+    ]
+  ]
+
+  skew_schema =
+    general ++
+      [
+        bias: [
+          type: :boolean,
+          default: true,
+          doc: "If false, then the calculations are corrected for statistical bias."
+        ]
+      ]
+
+  kurtosis_schema =
+    general ++
+      [
+        bias: [
+          type: :boolean,
+          default: true,
+          doc: "If false, then the calculations are corrected for statistical bias."
+        ],
+        variant: [
+          type: {:in, [:fisher, :pearson]},
+          default: :fisher,
+          doc:
+            "If :fisher then Fisher's definition is used, if :pearson then Pearson's definition is used."
+        ]
+      ]
+
+  @moment_schema NimbleOptions.new!(general)
+  @skew_schema NimbleOptions.new!(skew_schema)
+  @kurtosis_schema NimbleOptions.new!(kurtosis_schema)
+
+  @doc """
+  Calculates the nth moment about the mean for a sample.
+
+  ## Options
+
+  #{NimbleOptions.docs(@moment_schema)}
+
+  ## Examples
+
+      iex> x = Nx.tensor([[3, 5, 3], [2, 6, 1], [9, 3, 2], [1, 6, 8]])
+      iex> Scholar.Stats.moment(x, 2)
+      #Nx.Tensor<
+        f32[3]
+        [9.6875, 1.5, 7.25]
+      >
+  """
+  deftransform moment(tensor, moment, opts \\ []) do
+    opts = NimbleOptions.validate!(opts, @moment_schema)
+
+    num_samples =
+      Enum.product(Enum.map(opts[:axes] || Nx.axes(tensor), &Nx.axis_size(tensor, &1)))
+
+    moment_n(tensor, moment, num_samples, opts)
+  end
+
+  defnp moment_n(tensor, moment, num_samples, opts) do
+    mean = Nx.mean(tensor, axes: opts[:axes], keep_axes: true) |> Nx.broadcast(Nx.shape(tensor))
+    Nx.sum((tensor - mean) ** moment, opts) / num_samples
+  end
+
+  @doc """
+  Computes the sample skewness of a data set.
+
+  ## Options
+
+  #{NimbleOptions.docs(@skew_schema)}
+
+  ## Examples
+
+      iex> x = Nx.tensor([[3, 5, 3], [2, 6, 1], [9, 3, 2], [1, 6, 8]])
+      iex> Scholar.Stats.skew(x)
+      #Nx.Tensor<
+        f32[3]
+        [0.9794093370437622, -0.8164965510368347, 0.9220733642578125]
+      >
+  """
+
+  deftransform skew(tensor, opts \\ []) do
+    opts = NimbleOptions.validate!(opts, @skew_schema)
+
+    num_samples =
+      Enum.product(Enum.map(opts[:axes] || Nx.axes(tensor), &Nx.axis_size(tensor, &1)))
```

Comment:
Let's refactor this into a private function called `num_samples(tensor, opts)` and reuse it in the three definitions?